src/core/directives/google-map-polyline.ts

```diff
@@ -0,0 +1,224 @@
+
+  /**
+   * The zIndex compared to other polys.
+   */
+  zIndex: number;
+
+  /**
+   * This event is fired when the DOM click event is fired on the Polyline.
+   */
+  lineClick: EventEmitter<PolyMouseEvent> = new EventEmitter<PolyMouseEvent>();
+
+  /**
+   * This event is fired when the DOM dblclick event is fired on the Polyline.
+   */
+  lineDblClick: EventEmitter<PolyMouseEvent> = new EventEmitter<PolyMouseEvent>();
+
+  /**
+   * This event is repeatedly fired while the user drags the polyline.
+   */
+  lineDrag: EventEmitter<MouseEvent> = new EventEmitter<MouseEvent>();
+
+  /**
+   * This event is fired when the user stops dragging the polyline.
+   */
+  lineDragEnd: EventEmitter<MouseEvent> = new EventEmitter<MouseEvent>();
+
+  /**
+   * This event is fired when the user starts dragging the polyline.
+   */
+  lineDragStart: EventEmitter<MouseEvent> = new EventEmitter<MouseEvent>();
+
+  /**
+   * This event is fired when the DOM mousedown event is fired on the Polyline.
+   */
+  lineMouseDown: EventEmitter<PolyMouseEvent> = new EventEmitter<PolyMouseEvent>();
+
+  /**
+   * This event is fired when the DOM mousemove event is fired on the Polyline.
+   */
+  lineMouseMove: EventEmitter<PolyMouseEvent> = new EventEmitter<PolyMouseEvent>();
+
+  /**
+   * This event is fired on Polyline mouseout.
+   */
+  lineMouseOut: EventEmitter<PolyMouseEvent> = new EventEmitter<PolyMouseEvent>();
+
+  /**
+   * This event is fired on Polyline mouseover.
+   */
+  lineMouseOver: EventEmitter<PolyMouseEvent> = new EventEmitter<PolyMouseEvent>();
+
+  /**
+   * This event is fired whe the DOM mouseup event is fired on the Polyline
+   */
+  lineMouseUp: EventEmitter<PolyMouseEvent> = new EventEmitter<PolyMouseEvent>();
+
+  /**
+   * This even is fired when the Polyline is right-clicked on.
+   */
+  lineRightClick: EventEmitter<PolyMouseEvent> = new EventEmitter<PolyMouseEvent>();
+
+  @ContentChildren(SebmGoogleMapPolylinePoint) _points: QueryList<SebmGoogleMapPolylinePoint>;
+
+  private _id: string;
+  private _polylineAddedToManager: boolean = false;
+  private _observableSubscriptions: Subscription[] = [];
+
+  constructor(private _polylineManager: PolylineManager) { this._id = (polylineId++).toString(); }
+
+  /** @internal */
+  ngAfterContentInit() {
+    if (this._points.length) {
+      this._points.forEach((point: SebmGoogleMapPolylinePoint) => {
+        point.positionChanged.subscribe(
+            () => { this._polylineManager.updatePolylinePoints(this); });
+      });
+    }
+    if (!this._polylineAddedToManager) {
+      this._init();
+    }
+    // FIXME: once QueryList<> implements the Observable interface we can do this as needed
+    // FIXME: we are also limited to a fixed set of path items so far
+    this._polylineManager.updatePolylinePoints(this);
+  }
+
+  ngOnChanges(changes: SimpleChanges): any {
+    if (!this._polylineAddedToManager) {
+      this._init();
+      return;
+    }
+
+    if (changes['draggable']) {
```

Comment:
All these possible changes are in the `PolylineOptions`. Can you use mechanism like in `SebmGoogleMap`?
(1) https://github.com/SebastianM/angular2-google-maps/blob/master/src/core/directives/google-map.ts#L169
(2) https://github.com/SebastianM/angular2-google-maps/blob/master/src/core/directives/google-map.ts#L259

So we dont need explicit methods in the manager for every possible option change.